naga/tests/snapshots.rs

```diff
@@ -906,6 +916,36 @@ fn convert_wgsl() {
     }
 }
 
+#[cfg(feature = "wgsl-in")]
+#[test]
+fn unconsumed_vertex_outputs_hlsl_out() {
+    let load_and_parse = |name| {
+        // WGSL shaders lives in root dir as a privileged.
+        let input = Input::new(None, name, "wgsl");
+        let source = input.read_source();
+        let module = match naga::front::wgsl::parse_str(&source) {
+            Ok(module) => module,
+            Err(e) => panic!(
+                "{}",
+                e.emit_to_string_with_path(&source, input.input_path())
+            ),
+        };
+        (input, module)
+    };
+
+    // Uses separate wgsl files to make sure the tested code doesn't accidentally rely on
+    // the fragment entry point being from the same parsed content (e.g. accidentally using the
+    // wrong `Module` when looking up info). We also don't just create a module from the same file
+    // twice since everything would probably be stored behind the same keys.
+    let (input, mut module) = load_and_parse("unconsumed_vertex_outputs_vert");
+    let (frag_input, mut frag_module) = load_and_parse("unconsumed_vertex_outputs_frag");
```

Comment:
Adding this was a bit awkward since it didn't align very well with the existing tests. I needed a new parameter on `check_targets`. It looks like I could have added a new field to the `Parameters` struct to specify the fragment entry point. However, I wanted to specifically test the case of the fragment shader being from a separate `Module`.

Reply:
Thanks for the thorough tests!